src/icrc1_auction_api.mo

```diff
@@ -614,16 +629,24 @@ actor class Icrc1AuctionAPI(quoteLedger_ : ?Principal, adminPrincipal_ : ?Princi
   public shared ({ caller }) func registerAsset(ledger : Principal, minAskVolume : Nat) : async UpperResult<Nat, RegisterAssetError> {
     await* assertAdminAccess(caller);
     // validate ledger
-    let canister = actor (Principal.toText(ledger)) : (actor { icrc1_metadata : () -> async [Any] });
-    try {
-      ignore await canister.icrc1_metadata();
+    let canister = actor (Principal.toText(ledger)) : (actor { icrc1_decimals : () -> async Nat8 });
+    let decimals = try {
+      await canister.icrc1_decimals();
     } catch (err) {
       throw err;
     };
-    registerAsset_(ledger, minAskVolume) |> R.toUpper(_);
+    registerAsset_(ledger, minAskVolume, Prim.nat8ToNat(decimals)) |> R.toUpper(_);
   };
 
-  public shared query func debugLastBidProcessingInstructions() : async Nat64 = async lastBidProcessingInstructions;
+  public shared func updateDecimals() : async () {
+    for ((assetData, i) in Vec.items(assets)) {
+      let canister = actor (Principal.toText(assetData.ledgerPrincipal)) : (actor { icrc1_decimals : () -> async Nat8 });
+      let decimals = (await canister.icrc1_decimals()) |> Prim.nat8ToNat(_);
+      if (assetData.decimals != decimals) {
+        Vec.put(assets, i, { assetData with decimals });
+      };
+    };
+  };
```

Comment:
We can skip this function. Decimals of an icrc-1 ledger must never change.

Reply:
I need this at least once after upgrade. I'll run it once after deployment and then remove

Reply:
Removed